wp-cache-phase1.php

```diff
@@ -632,22 +674,16 @@ function get_current_url_supercache_dir( $post_id = 0 ) {
 			 * Sometimes site_url doesn't return the siteurl. See http://wordpress.org/support/topic/wp-super-cache-not-refreshing-post-after-comments-made
 			*/
 			$DONOTREMEMBER = 1;
-			wp_cache_debug( "get_current_url_supercache_dir: warning! site_url ($site_url) not found in permalink ($permalink).", 1 );
-			if ( false === strpos( $permalink, $WPSC_HTTP_HOST ) ) {
-				wp_cache_debug( "get_current_url_supercache_dir: WARNING! SERVER_NAME ({$WPSC_HTTP_HOST}) not found in permalink ($permalink). ", 1 );
-				$p = parse_url( $permalink );
-				if ( is_array( $p ) ) {
-					$uri = $p[ 'path' ];
-					wp_cache_debug( "get_current_url_supercache_dir: WARNING! Using $uri as permalink. Used parse_url.", 1 );
-				} else {
-					wp_cache_debug( "get_current_url_supercache_dir: WARNING! Permalink ($permalink) could not be understood by parse_url. Using front page.", 1 );
-					$uri = '';
+			wp_cache_debug( "get_current_url_supercache_dir: WARNING! site_url ($site_url) not found in permalink ($permalink).", 1 );
+			if ( preg_match( '#^(https?:)?//([^/]+)(/[^\?\#]*)?((\?|\#).*)?$#i', $permalink, $matches ) && !empty( $matches[2] ) ) {
```

Comment:
Probably, it's possible to reduce complexity of this regexp. It's based on WP core examples and my experience. Anyway, query string is removed later in the code. Fragment identifier maybe we should remove (I've one website which makes these URLs).